File: QuizPortal-Server/src/main/java/com/QuizPortalServer/QuizPortalServer/service/QuizService.java
package com.QuizPortalServer.QuizPortalServer.service;

import com.QuizPortalServer.QuizPortalServer.model.quiz.Category;
import com.QuizPortalServer.QuizPortalServer.model.quiz.Quiz;

import java.util.List;

public interface QuizService {
    public Quiz AddQuiz(Quiz quiz);
    public List<Quiz> getQuiz();
    public Quiz getQuizById(long quizId);
    public Quiz updateQuiz(Quiz quiz);
    public void deleteQuiz(Long quizId);
    public List<Quiz> getQuizzesOfCategory(Category category);
    public List<Quiz> getActiveQuizzes();
    public List<Quiz> getActiveQuizzesOfCategory(Category category);
}

File: QuizPortal-Client/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http : HttpClient) { }

  getAllQuiz(){
    return this.http.get(environment.baseUrl + '/quiz');
  }

  getQuizById(quizId :any){
    return this.http.get(`${environment.baseUrl}/quiz/${quizId}`)
  }

  createQuiz(data :any){
    return this.http.post(environment.baseUrl + '/quiz',data)
  }

  updateQuiz(data :any){
    return this.http.put(environment.baseUrl + '/quiz',data)
  }


  removeQuiz(quizId :any){
    return this.http.delete(`${environment.baseUrl}/quiz/${quizId}`)
  }

  getQuizzesOfCategories(categoryId :any){
    return this.http.get(`${environment.baseUrl}/quiz/category/${categoryId}`)
  }
    //get active quizzes
  getActiveQuizzes(){
    return this.http.get(environment.baseUrl + '/quiz/active')
  }
    // get Active Quizzes Of Category
  getActiveQuizzesOfCategory(categoryId :any){
    return this.http.get(`${environment.baseUrl}/quiz/category/active/${categoryId}`)
  }
}
